larq_compute_engine/mlir/transforms/prepare_patterns_common.td

```diff
@@ -4,9 +4,24 @@ include "tensorflow/compiler/mlir/tensorflow/ir/tf_ops.td"
 include "larq_compute_engine/mlir/ir/lce_ops.td"
 include "larq_compute_engine/mlir/transforms/op_removal_patterns.td"
 
+class ConstantValue<string val> : AttrConstraint<CPred<"IsConstantValue($_self, " # val # ")">>;
 
-// This relies on implementation details of larq.math.sign. We should make
-// this more general in the future
+// Base quantiser patterns that match the `tf.where` implementation of `ste_sign`.
+def : Pat<(TF_SelectV2Op:$select_op
```

Comment:
I agree we should probably also match the v1 op :+1:

With respect to shapes, the pattern correctly works when the lhs or rhs is lower rank than the input (there's an MLIR test for this). I see what you mean though, that if the condition is lower rank than the lhs or rhs then we need to broadcast it. Do you think it would be easiest to insert a broadcast op then?